ui/apps/platform/src/Components/CompoundSearchFilter/attributes/imageCVE.ts

```diff
@@ -23,4 +23,51 @@ export const CVSS: CompoundSearchFilterAttribute = {
     inputType: 'condition-number',
 };
 
-export const imageCVEAttributes = [Name, DiscoveredTime, CVSS];
+export const EPSSProbability: ConditionTextFilterAttribute = {
+    displayName: 'EPSS probability',
+    filterChipLabel: 'EPSS probability',
+    searchTerm: 'EPSS Probability',
+    inputType: 'condition-text',
+    featureFlagDependency: ['ROX_SCANNER_V4', 'ROX_EPSS_SCORE'],
+    inputProps: {
+        conditionProps: {
+            conditionEntries: [
+                ['>', 'Is greater than'],
+                ['>=', 'Is greater than or equal to'],
+                ['=', 'Is equal to'],
+                ['<=', 'Is less than or equal to'],
+                ['<', 'Is less than'],
+            ],
+        },
+        textProps: {
+            convertFromExternalToInternalText: (externalText: string) => {
+                const float = parseFloat(externalText); // assume validateExternalText
+                return String(float / 100); // from percent to fraction
+            },
+            convertFromInternalToExternalText: (internalText: string) => {
+                const float = parseFloat(internalText); // assume validateInternalText
+                const percent = float * 100; // from fraction to percent
+                return `${percent.toFixed(3)}%`; // same precison as EPSS data
+            },
+            externalTextDefault: '0%',
+            validateExternalText: (externalText: string) => {
+                if (!/^(\.\d+|\d+(?:\.\d*)?)%?$/.test(externalText.trim())) {
+                    return false;
+                }
+                const float = parseFloat(externalText);
+                return !Number.isNaN(float) && float >= 0 && float <= 100;
+            },
+            validateInternalText: (internalText: string) => {
+                // Assume internalText was serialized by String (see above).
+                // Leading zero followed by optional decimal point and digits.
+                if (!/^(\.\d+|\d+(?:\.\d*)?)$/.test(internalText.trim())) {
+                    return false;
+                }
+                const float = parseFloat(internalText);
+                return !Number.isNaN(float) && float >= 0 && float <= 1;
+            },
```

Comment:
Certainly not a request for change, but what do you think are the downsides of performing validation _only_ on the "internal" version of the value?

That is, we validate on incoming data, and outgoing data, and attempt to parse a valid value from user entered text before committing the value to the search filter (refusing to do so if the parsing fails). That way we skip the need to validate the external format, as we do the validation only on the internally used value?

Reply:
A thought for the slow cooker. We have time to measure twice or thrice, and then commit once.

Reply:
Now that my eyes rested on the code, I see that that code comment is obsolete and even misleading.

All the more reason for unit tests.